src/nacl/bindings/crypto_pwhash_scryptsalsa208sha256.py

```diff
@@ -0,0 +1,145 @@
+        - :py:const:`.MEMLIMIT_INTERACTIVE`
+        - :py:const:`.OPSLIMIT_SENSITIVE`
+        - :py:const:`.MEMLIMIT_SENSITIVE`
+
+    are provided as a guidance for correct settings respectively for the
+    interactive login and the long term key protecting sensitive data
+    usage cases.
+
+    :param int outlen: int
+    :param bytes passwd: bytes
+    :param bytes salt: *must* be *exactly* :py:const:`.SALTBYTES` long
+    :param int opslimit:
+    :param int memlimit:
+    :rtype: bytes
+    """
+
+    ensure(len(salt) == SALTBYTES, 'Invalid salt',
+           raising=exc.ValueError)
+
+    buf = ffi.new("unsigned char[]", outlen)
+
+    ret = lib.crypto_pwhash_scryptsalsa208sha256(buf, outlen, passwd,
+                                                 len(passwd), salt,
+                                                 opslimit, memlimit)
+
+    ensure(ret == 0, 'Unexpected failure in key derivation',
+           raising=exc.RuntimeError)
+
+    return ffi.buffer(buf, outlen)[:]
+
+
+def crypto_pwhash_scryptsalsa208sha256_str(passwd,
+                                           opslimit=OPSLIMIT_INTERACTIVE,
+                                           memlimit=MEMLIMIT_INTERACTIVE
+                                           ):
+    """
+    Derive a cryptographic key using the ``passwd`` and ``salt``
+    given as input, returning a string representation which includes
+    the salt and the tuning parameters.
+
+    The returned string can be directly stored as a password hash.
+
+    See :py:func:`.crypto_pwhash_scryptsalsa208sha256` for a short
+    discussion about ``opslimit`` and ``memlimit`` values.
+
+    :param bytes passwd:
+    :param int opslimit:
+    :param int memlimit:
+    :return: serialized key hash, including salt and tuning parameters
+    :rtype: bytes
+    """
+    buf = ffi.new("unsigned char[]", STRBYTES)
+
+    ret = lib.crypto_pwhash_scryptsalsa208sha256_str(buf, passwd,
+                                                     len(passwd),
+                                                     opslimit,
+                                                     memlimit)
+
+    ensure(ret == 0, 'Unexpected failure in password hashing',
+           raising=exc.RuntimeError)
+
+    return ffi.string(buf)
+
+
+def crypto_pwhash_scryptsalsa208sha256_str_verify(passwd_hash, passwd):
+    """
+    Verifies the ``passwd`` against the ``passwd_hash`` that was generated.
+    Returns True or False depending on the success
+
+    :param passwd_hash: bytes
+    :param passwd: bytes
+    :rtype: boolean
+    """
+
+    ensure(len(passwd_hash) == STRBYTES - 1, 'Invalid password hash',
+           raising=exc.ValueError)
+
+    ret = lib.crypto_pwhash_scryptsalsa208sha256_str_verify(passwd_hash,
+                                                            passwd,
+                                                            len(passwd))
+    ensure(ret == 0,
+           "Wrong password",
+           raising=exc.ValueError)
```

Comment:
I think this should be a separate exception like InvalidKey rather than the same exception you'd get from a wrong length.